### PR TITLE
feat: id stitcher report now has the graph browser

### PR DESCRIPTION
**Pull request by nishantsharma**
## Description of the change
Check out inputs.yaml also. In IdStitcher Report, user can now pass an explicit list of clusters(by cluster main ID, or by any its member IDs).

1. Top 10 clusters by size are retrieved.
2. Top 10 clusters by (approx) diameter are retrieved.
3. Clusters in user supplied clusters' list CSV are also retrieved.

All the listed clusters can be graphically browsed.

### Demo
https://rudderlabs.slack.com/archives/C06757TF6KZ/p1725291234778129?thread_ts=1724823275.987379&cid=C06757TF6KZ
https://rudderlabs.slack.com/files/U02M8RGUHMX/F07K8MQTQ23/material_id_stitcher_report_76a45b8f_41.zip

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

https://linear.app/rudderstack/issue/PRO-3369/add-graph-visualisation-in-id-stitcher-report
https://linear.app/rudderstack/issue/PRO-3370/default-folder-for-m=thisderefpathref-should-be-level-root-while-for

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by SameerPr — APPROVED** [review]

**Review by sprksh — APPROVED** [review]
Didnt dive deep into sql changes. LGTM OW.

File: models/profiles.yaml [hunk 1]
```diff
@@ -35,18 +33,17 @@ models:
...
       single_sql: |
-        {% with input_material = this.DeRef("inputs/id_stitcher") %}
-          {% if input_material.Model.ModelType() == "id_stitcher" %}
-            select count(distinct(node_id)) as id_count, max({{input_material.Model.GetEntity().IdColumnName}}_dist) as max_dist, {{input_material.Model.GetEntity().IdColumnName}}
-            from {{input_material}}_internal_mapping
-            group by {{input_material.Model.GetEntity().IdColumnName}}
-          {% endif %}
+        {% with idstitcher = this.DeRef("inputs/id_stitcher") %}
+            {% if idstitcher.Model.ModelType() == "id_stitcher" %}
+                {% with ids_mappings = idstitcher.DeRef("mappings") %}
```
Comment (nishantsharma) [inline models/profiles.yaml]: ./mappings

File: models/profiles.yaml [hunk 2]
```diff
@@ -91,123 +78,170 @@ models:
...
-              )
-              group by m.{{input_material.Model.GetEntity().IdColumnName}}, e.node_id, e.node_id_type
-              order by degree desc
-            ) as ranked
-            where rn <= 10
-          {% endif %}
+        {% with idstitcher = this.DeRef("inputs/id_stitcher") %}
+            {% if idstitcher.Model.ModelType() == "id_stitcher" %}
+                {% with id_column_name = idstitcher.Model.GetEntity().IdColumnName %}
+                {% with ids_edges = idstitcher.DeRef("edges") %}
```
Comment (nishantsharma) [inline models/profiles.yaml]: ./edges

File: models/profiles.yaml [hunk 3]
```diff
@@ -91,123 +78,170 @@ models:
...
-            from {{input_material1}}_internal_edges A inner join {{input_material1}} B on A.id1 = B.other_id
-            where id1 <> id2 AND id1 in (select node_id from {{input_material2.Name()}}) AND id2 in (select node_id from {{input_material2.Name()}})
-          {% endif %}
+        {% with idstitcher = this.DeRef("inputs/id_stitcher") %}
+        {% with seed_clusters_optional = this.DeRef("models/seed_clusters_processed", dependency="optional") %}
+        {% with clusters_sorted_by_size = this.DeRef("models/clusters_sorted_by_size") %}
+        {% with clusters_sorted_by_dist = this.DeRef("models/clusters_sorted_by_dist") %}
+            {% if idstitcher.Model.ModelType() == "id_stitcher" %}
+                {% with id_column_name = idstitcher.Model.GetEntity().IdColumnName %}
+                {% with ids_edges = idstitcher.DeRef("edges") %}
```
Comment (nishantsharma) [inline models/profiles.yaml]: ./edges

File: models/profiles.yaml [hunk 4]
```diff
@@ -91,123 +78,170 @@ models:
...
-            from {{input_material1}} A inner join {{input_material2.Name()}} B on A.{{input_material1.Model.GetEntity().IdColumnName}} = B.{{input_material1.Model.GetEntity().IdColumnName}}
-            group by A.{{input_material1.Model.GetEntity().IdColumnName}}, other_id_type
-            order by total_id_count desc , id_type_count desc
-          {% endif %}
+      single_sql: |-
+        {% with idstitcher = this.DeRef("inputs/id_stitcher") %}
+        {% with clusters_sorted_by_size = this.DeRef("models/clusters_sorted_by_size") %}
+            {% if idstitcher.Model.ModelType() == "id_stitcher" %}
+                {% with id_column_name = idstitcher.Model.GetEntity().IdColumnName %}
+                {% with ids_edges = idstitcher.DeRef("edges") %}
```
Comment (nishantsharma) [inline models/profiles.yaml]: ./edges

**Pull request merged**